File: 0007-reverse-integer/0007-reverse-integer.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    let sign = x > 0 ? 1 : -1;
    let revNum = 0;
    let num = Math.abs(x);
    while(num > 0){
        let rem = num % 10;
        revNum = revNum * 10 + rem;
        num = Math.floor(num/10);
    }
    revNum = revNum * sign;
     if (revNum < -2147483648 || revNum > 2147483647) {
        return 0;
    }
    return revNum;
};

File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        int sign = x > 0 ? 1 : -1;
        long revNo = 0;
        int num = Math.abs(x);
        while(num > 0){
            int rem = num % 10;
            revNo = revNo * 10 + rem;
            num = num / 10; 
        }
        revNo = revNo * sign;
        if(revNo < Integer.MIN_VALUE || revNo > Integer.MAX_VALUE){
            return 0;
        }
        return (int)revNo;
    }  
}